src/runtime_src/xdp/profile/writer/vp_base/vp_writer.cpp

```diff
@@ -28,10 +36,61 @@ namespace xdp {
   {
   }
 
-  VPWriter::VPWriter(const char* filename, VPDatabase* inst) :
-    basename(filename), currentFileName(filename), fileNum(1), db(inst),
-    fout(filename)
+  VPWriter::VPWriter(const char* filename, VPDatabase* inst, bool useDir) :
+    basename(filename), currentFileName(filename), directory(""),
+#ifdef _WIN32
+    separator('\\'),
+#else
+    separator('/'),
+#endif
+    fileNum(1), db(inst)
   {
+    directory = xrt_core::config::get_profiling_directory() ;
+
+    if (!useDir || directory == "") {
+      // If no directory was specified, just use the file in
+      //  the working directory
+      fout.open(filename) ;
+      return ;
+    }
+
+    // The directory was specified.  Check if it exists and is writable
+    bool dirExists  = false ;
+    bool writeable  = false ;
+#ifdef _WIN32
+#else
+    struct stat buf ;
+    int result = stat(directory.c_str(), &buf) ;
+    if (!result) {
+      // The file exists.  Is it a directory?
+      dirExists = S_ISDIR(buf.st_mode) ;
+      if (dirExists) {
+        // The directory exists, but can we write to it?
+        writeable = (buf.st_mode & S_IWUSR) != 0 ;
+      }
+      //else {
+        // The file exists, but it is not a directory.  We cannot write to it
+      //}
+    }
+    else {
+      // The file does not exist at all, so try to create it
+      result = mkdir(directory.c_str(), 0777) ;
```

Comment:
warning: 0777 is a magic number; consider replacing it with a named constant [cppcoreguidelines-avoid-magic-numbers]

```cpp
result = mkdir(directory.c_str(), 0777) ;
                                  ^
```

Reply:
Added named constant rwx_all for the 0777 value in latest commit.